### ISSUE TITLE
Tags

### ISSUE THREAD

**Issue by jantimon** [htmllint/htmllint#174]
If you use https://validator.w3.org/check#source to lint the following code it will tell you that a div must not be inside a span:

```html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head><title>Demo</title></head>
<body>
<span><div></div></span>
</body>
</html>
```

@mlochbaum Would you be open to a pull request which adds this feature also for htmllint?

**Comment by mlochbaum** [comment]
Of course! Kind of a hard one to name--maybe `div-inside-span-ban` makes sense?

`fig-ref-figcaption` is a little similar if you want a reference to look at.

**Comment by jantimon** [comment]
I would call it xhtml-strict:true and use a npm like https://www.npmjs.com/package/libxml-xsd to validate the html against https://www.w3.org/2002/08/xhtml/xhtml1-strict.xsd

**Comment by jantimon** [comment]
After further research I found that the official html5 validator settings are here: https://github.com/validator/validator/tree/master/schema/html5
However I couldn't find a way to validate against them using javascript
